superset-frontend/src/SqlLab/components/SqlEditor/index.jsx

```diff
@@ -140,172 +139,158 @@ const StyledSidebar = styled.div`
     // to smooth out the visual transition to the new heights when drag ends
     document.getElementsByClassName('ace_content')[0].style.height = '100%';
-  }
+  };
 
-  onResizeEnd([northPercent, southPercent]) {
-    this.setState({ northPercent, southPercent });
+  const onResizeEnd = ([northPercent, southPercent]) => {
+    setNorthPercent(northPercent);
+    setSouthPercent(southPercent);
 
-    if (this.northPaneRef.current && this.northPaneRef.current.clientHeight) {
-      this.props.persistEditorHeight(
-        this.props.queryEditor,
-        northPercent,
-        southPercent,
-      );
+    if (northPaneRef.current && northPaneRef.current.clientHeight) {
+      dispatch(persistEditorHeight(queryEditor, northPercent, southPercent));
     }
-  }
+  };
 
-  onBeforeUnload(event) {
+  const onBeforeUnload = event => {
     if (
-      this.props.database?.extra_json?.cancel_query_on_windows_unload &&
-      this.props.latestQuery?.state === 'running'
+      database?.extra_json?.cancel_query_on_windows_unload &&
+      latestQuery?.state === 'running'
     ) {
       event.preventDefault();
-      this.stopQuery();
+      stopQuery();
     }
-  }
+  };
 
-  onSqlChanged(sql) {
-    this.props.queryEditorSetSql(this.props.queryEditor, sql);
-    this.setQueryEditorAndSaveSqlWithDebounce(sql);
+  const onSqlChanged = sql => {
+    dispatch(queryEditorSetSql(queryEditor, sql));
+    setQueryEditorAndSaveSqlWithDebounce(sql);
     // Request server-side validation of the query text
-    if (this.canValidateQuery()) {
+    if (canValidateQuery()) {
       // NB. requestValidation is debounced
-      this.requestValidation(sql);
+      requestValidationWithDebounce(sql);
     }
-  }
+  };
 
   // One layer of abstraction for easy spying in unit tests
-  getSqlEditorHeight() {
-    return this.sqlEditorRef.current
-      ? this.sqlEditorRef.current.clientHeight - SQL_EDITOR_PADDING * 2
+  const getSqlEditorHeight = () =>
+    sqlEditorRef.current
+      ? sqlEditorRef.current.clientHeight - SQL_EDITOR_PADDING * 2
       : 0;
-  }
 
   // Return the heights for the ace editor and the south pane as an object
   // given the height of the sql editor, north pane percent and south pane percent.
-  getAceEditorAndSouthPaneHeights(height, northPercent, southPercent) {
-    return {
-      aceEditorHeight:
-        (height * northPercent) / 100 -
-        (SQL_EDITOR_GUTTER_HEIGHT / 2 + SQL_EDITOR_GUTTER_MARGIN) -
-        SQL_TOOLBAR_HEIGHT,
-      southPaneHeight:
-        (height * southPercent) / 100 -
-        (SQL_EDITOR_GUTTER_HEIGHT / 2 + SQL_EDITOR_GUTTER_MARGIN),
-    };
-  }
-
-  getHotkeyConfig() {
+  const getAceEditorAndSouthPaneHeights = (
+    height,
+    northPercent,
+    southPercent,
+  ) => ({
+    aceEditorHeight:
+      (height * northPercent) / 100 -
+      (SQL_EDITOR_GUTTER_HEIGHT / 2 + SQL_EDITOR_GUTTER_MARGIN) -
+      SQL_TOOLBAR_HEIGHT,
+    southPaneHeight:
+      (height * southPercent) / 100 -
```

Comment:
```suggestion
      (height * northPercent) / (theme.gridUnit * 25) -
      (SQL_EDITOR_GUTTER_HEIGHT / 2 + SQL_EDITOR_GUTTER_MARGIN) -
      SQL_TOOLBAR_HEIGHT,
    southPaneHeight:
      (height * southPercent) / (theme.gridUnit * 25) -
```
Unit values should stay within the Superset theme